packages/react/src/components/Table/TableHead/TableHead.jsx

```diff
@@ -428,9 +469,31 @@ const TableHead = ({
     // since it would be directly overridden by the column props. This effect can be removed
     // with issue https://github.com/IBM/carbon-addons-iot-react/issues/1224
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [hasResize, columns, ordering, previousColumns]
+    [hasResize, columns, ordering, previousColumns, percentageMode]
   );
 
+  useEffect(() => {
+    if (enablePercentageColumnWidth && columns.length && percentageMode) {
+      const width = `${parseInt(100 / columns.length, 10)}`;
+
+      const updatedColumns = columns.map((col) => {
+        const updatedCol = { ...col };
+        updatedCol.width = `${width}%`;
```

Comment:
The PAL table does not do a good job with default width(pixel width) even we have enough space left in the table. It should, by default, provide a % column width based on how many columns there are based on a props. And when we will pass that props there will not be any fixed pixel column width, all columns will have same % width.
1 column = 100%
2 columns = 50/50
3 columns = 33/33/33

Reply:
The user will not be able to specify a fixed percentage width for any column; instead, it will be calculated automatically based on the number of columns present in the table.